test/test_pooledembeddings.jl

```diff
@@ -0,0 +1,81 @@
+using Polynomials4ML: legendre_basis, RYlmBasis, ScalarPoly4MLBasis, PooledSparseProduct, OrthPolyBasis1D3T, PooledEmbeddings
+using StaticArrays, LinearAlgebra
+using ObjectPools: acquire!, release!
+using Polynomials4ML
+using ACEbase.Testing: fdtest, print_tf
+using Test
+using Printf
+using Random
+using Lux
+using Zygote
+
+function _generate_basis(; order=2, len = 50)
+   NN = [ rand(5:10) for _ = 1:order ]
+   spec = sort([ ntuple(t -> rand(1:NN[t]), order) for _ = 1:len])
+   return PooledSparseProduct(spec)
+end
+
+Rnl = legendre_basis(10)
+Ylm = RYlmBasis(10)
+
+pooling = _generate_basis()
+
+X = [ @SVector(randn(3)) for i in 1:3 ]
+
+embeddings = (Rnl, Ylm)
+embed_and_pool = Polynomials4ML.PooledEmbeddings(embeddings, pooling)
```

Comment:
I think this is good for now, but I think we may want a second constructor of the form 
```julia
PooledEmbeddings(embeddings, spec)
```